### PR TITLE
add support for specifying the name of the endpoint

### PR DESCRIPTION
**Pull request by lurock**
You add a "traefik.portName" label to your Service Fabric manifest and specify the name of the endpoint that you want to serve web traffic through.

Example:

```xml
<?xml version="1.0" encoding="utf-8"?>
<ServiceManifest Name="WebPkg" Version="1.0.0" xmlns="http://schemas.microsoft.com/2011/01/fabric" xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
  <ServiceTypes>
    <!-- This is the name of your ServiceType.
         The UseImplicitHost attribute indicates this is a guest service. -->
    <StatelessServiceType ServiceTypeName="WebType" UseImplicitHost="true">
      <Extensions>
        <Extension Name="Traefik">
          <Labels xmlns="http://schemas.microsoft.com/2015/03/fabact-no-schema">
            <Label Key="traefik.servicefabric.endpointname">DefaultEndpoint</Label>
            <Label Key="traefik.backend.healthcheck.path">/</Label>
            <Label Key="traefik.backend.healthcheck.port">5050</Label>
            <Label Key="traefik.backend.healthcheck.interval">5s</Label>
            <Label Key="traefik.frontend.priority">10</Label>
            <Label Key="traefik.enable">true</Label>
          </Labels>
        </Extension>
      </Extensions>
    </StatelessServiceType>
  </ServiceTypes>

  <!-- Code package is your service executable. -->
  <CodePackage Name="Code" Version="1.0.0">
      <EntryPoint>
      <ExeHost>
        <Program>Web.exe</Program>
        <WorkingFolder>CodeBase</WorkingFolder>
        <!-- Uncomment to log console output (both stdout and stderr) to one of the
             service's working directories. -->
        <ConsoleRedirection FileRetentionCount="5" FileMaxSizeInKb="2048"/>
      </ExeHost>
    </EntryPoint>
  </CodePackage>
  <ConfigPackage Name="Config" Version="1.0.0" />

  <Resources>
    <Endpoints>
      <!-- This endpoint is used by the communication listener to obtain the port on which to
           listen. Please note that if your service is partitioned, this port is shared with
           replicas of different partitions that are placed in your code. -->
      <Endpoint Name="DefaultEndpoint" UriScheme="http" Port="8000" Protocol="http" Type="Input" />
      <Endpoint Name="HealthCheckEndpoint" UriScheme="http" Port="5050" Protocol="http" Type="Input" />
    </Endpoints>
  </Resources>
</ServiceManifest>
```

### CODE REVIEW

**Comment by jjcollinge** [comment]
@lurock thanks, it looks good at first glance :). Would it be possible to get some test coverage for this too please?

**Comment by lawrencegripper** [comment]
@lurock Thanks for the PR! This makes sense to me, one tweak I'd propose is maybe moving the label to `traefik.servicefabric.endpointname`. 

My thinking is that [existing SF specific labels](https://github.com/containous/traefik-extra-service-fabric/blob/master/servicefabric_labelfuncs.go#L11) sit under `.servicefabric` to help users understand they'll be under the SF docs not the generic Traefik docs and as the `name` field we're referencing is on the `endpoint` object in SF I think it would fits better.

**Comment by lawrencegripper** [comment]
If you want any pointers on this drop onto the Traefik slack and there is a service-fabric channel, happy to chat/help over IM if that is easier.

**Comment by lurock** [comment]
I have been running it in our production cluster for over a week now and it has been working great.

File: servicefabric_tmpl.go [hunk 1]
```diff
@@ -64,22 +71,36 @@ const tmpl = `
         {{end}}
 
         {{range $instance := $partition.Instances}}
-          [backends."{{ $service.Name }}".servers."{{ $instance.ID }}"]
-            url = "{{ getDefaultEndpoint $instance }}"
-            weight = {{ getWeight $service }}
+          {{ $endpointName := getLabelValue $service "traefik.servicefabric.endpointname" "" }}
```
Comment (ldez) [inline servicefabric_tmpl.go]:
```
[backends."{{ $service.Name }}".servers."{{ $instance.ID }}"]
  weight = {{ getWeight $service }}
  {{ $endpointName := getLabelValue $service "traefik.servicefabric.endpointname" "" }}
  {{if $endpointName }}
    url = "{{ getNamedEndpoint $instance $endpointName }}"
  {{else}}
    url = "{{ getDefaultEndpoint $instance }}"
  {{end}}
```

File: servicefabric_tmpl.go [hunk 2]
```diff
@@ -64,22 +71,36 @@ const tmpl = `
...
 
       {{else if isStateful $service}}
 
         {{range $replica := $partition.Replicas}}
           {{if isPrimary $replica}}
             {{ $backendName := getBackendName $service $partition }}
-            [backends."{{ $backendName }}".servers."{{ $replica.ID }}"]
-              url = "{{ getDefaultEndpoint $replica }}"
-							weight = 1
+            {{ $endpointName := getLabelValue $service "traefik.servicefabric.endpointname" "" }}
```
Comment (ldez) [inline servicefabric_tmpl.go]:
```

[backends."{{ $backendName }}".servers."{{ $replica.ID }}"]
  weight = 1
  {{ $endpointName := getLabelValue $service "traefik.servicefabric.endpointname" "" }}
  {{if $endpointName }}
    url = "{{ getNamedEndpoint $replica $endpointName }}"
  {{else}}
    url = "{{ getDefaultEndpoint $replica }}"
  {{end}}
 ```

**Pull request merged**